test/e2e/registration_service_test.go

```diff
@@ -429,6 +431,177 @@ func (s *registrationServiceTestSuite) TestSignupOK() {
 	assert.Equal(s.T(), ExpectedConsoleURL(s.T(), s.memberAwait, memberCluster), mp["consoleURL"])
 }
 
+func (s *registrationServiceTestSuite) TestPhoneVerification() {
+	// Create a token and identity to sign up with
+	identity0 := authsupport.NewIdentity()
+	emailValue := uuid.NewV4().String() + "@some.domain"
+	emailClaim0 := authsupport.WithEmailClaim(emailValue)
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0)
+	require.NoError(s.T(), err)
+
+	// Call the signup endpoint
+	req, err := http.NewRequest("POST", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+	resp, err := httpClient.Do(req)
+	require.NoError(s.T(), err)
+	defer close(s.T(), resp)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+	assert.Equal(s.T(), http.StatusAccepted, resp.StatusCode)
+
+	// Wait for the UserSignup to be created
+	userSignup, err := s.hostAwait.WaitForUserSignup(identity0.ID.String(), wait.UntilUserSignupHasConditions(PendingApproval()...))
+	require.NoError(s.T(), err)
+	emailAnnotation := userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey]
+	assert.Equal(s.T(), emailValue, emailAnnotation)
+
+	// Call get signup endpoint with a valid token and make sure it's pending approval
+	req, err = http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+
+	resp, err = httpClient.Do(req)
+	require.NoError(s.T(), err)
+	defer close(s.T(), resp)
+
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+
+	mp := make(map[string]interface{})
+	err = json.Unmarshal([]byte(body), &mp)
+	require.NoError(s.T(), err)
+
+	mpStatus, ok := mp["status"].(map[string]interface{})
+	assert.True(s.T(), ok)
+
+	assert.Equal(s.T(), "", mp["compliantUsername"])
+	assert.Equal(s.T(), identity0.Username, mp["username"])
+	require.IsType(s.T(), false, mpStatus["ready"])
+	assert.False(s.T(), mpStatus["ready"].(bool))
+	assert.Equal(s.T(), "PendingApproval", mpStatus["reason"])
+	require.True(s.T(), mpStatus["verificationRequired"].(bool))
+
+	// Approve usersignup.
+	userSignup.Spec.Approved = true
+	err = s.hostAwait.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+
+	// Ensure the Master User Record is NOT provisioned
+	_, err = s.hostAwait.WaitForMasterUserRecord(identity0.Username)
+	require.Error(s.T(), err)
```

Comment:
We are waiting for the default timeout here. So, this test will be always too slow.
Let's maybe do the following instead:
- Wait for StatusVerificationRequired: https://github.com/codeready-toolchain/host-operator/blob/9dc20a3c6c4522f916dd7db045421e7d3bc638bf/pkg/controller/usersignup/usersignup_controller.go#L281
- Check (without wait) that no MUR created/exist

Reply:
For checking that it's not created you could use:
```
        err := s.hostAwait.WaitUntilMasterUserRecordDeleted(identity0.Username)
        require.NoError(t, err)
```

Reply:
`WaitUntilMasterUserRecordDeleted()` is good for waiting for the MUR to be deleted:
T1. MUR exists.
T2. Do something which should delete the MUR.
T3. Wait for the MUR to not exist.

But it won't work properly for making sure that the MUR has never created:
T1. MUR does not exist.
T2. Do something which should prevent the MUR to be created
T3. Wait for the MUR to not exist (can pass even if the MUR is created with a delay).

Reply:
I've already updated this code to what Matous suggested - is it the consensus that this won't actually work?

Reply:
`s.hostAwait.WaitUntilMasterUserRecordDeleted(identity0.Username) is unfortunately useless in this case. Take a look at my  previous message. So I think we have to implement it as a suggested in https://github.com/codeready-toolchain/toolchain-e2e/pull/167#discussion_r495219691.

Reply:
Yeah, I thought that checking that the MUR does not exist should be enough - for this particular use-case the method is sufficient - it verifies that he MUR does not exist. I know that ideally we should wait some short time and check that MUR is not really created, but since we are going to wait for StatusVerificationRequired (as suggested by Alexey), then we should be fine.

Reply:
There is a couple of cases where we don't have a choice and have to wait for some resource to be not be created. For example in https://github.com/codeready-toolchain/toolchain-e2e/blob/4996af7b68436dfe74338ef560c8e0cb60804060/test/e2e/usersignup_test.go#L133
But we introduced `WithRetryOptions()` to override the default timeout (which is too long) for such cases:
```
s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(username)
require.Error(s.T(), err)
```
It still slows down the test but not for the entire default timeout (10 sec vs 60 sec per call). But in this particular case I think we could just wait for StatusVerificationRequired and don't wait for MUR at all. So we can save those 10 seconds.

Reply:
I've removed the MUR check here and replaced it with a UserSignup status check, plus a check the confirm that `verificationRequired` is correctly set to true when calling the getSignup endpoint

Reply:
But we still need to check that the MUR is not created. Why not checking that it does not exist?

Reply:
Your very last comment was "But in this particular case I think we could just wait for StatusVerificationRequired and don't wait for MUR at all." What did you mean by this?

Reply:
If we wait for StatusVerificationRequired then we don't have to **wait** for the MUR to. In other words we don't have to use any `WaitFor...()` function and can check if the MUR exists or not without any timeout/retries/waits. Sorry for not being clear.

Reply:
Just try to load the MUR from the client. It should not found any.

Reply:
No problem, your use of the word "just" in that context made me think you only cared about the status check.  I'll add a lookup then for the MUR and expect an error there.